JS/Lesson33/gulcin/my-app/src/modules/list/ListProvider.jsx

```diff
@@ -0,0 +1,54 @@
+import { createContext, useReducer, useContext } from "react";
+
+
+export const todoData = [
+    {id: 1, title: "Do dishes", completed: false},
+    {id: 2, title: "Do homework", completed: false},
+    {id: 3, title: "Go running", completed: false},
+    {id: 4, title: "Do dishes 2", completed: false},
+    {id: 5, title: "Do homework 3", completed: false},
+    {id: 6, title: "Go running 4", completed: false},
+  ];
+
+export const TODOLIST_ACTIONS = {
+   toDoCompleted: 'todo_completed',
+   delete: 'delete',
+
+}
+
+export const TodoListContext = createContext();
+export const TodoListDispatchContext = createContext();
+
+export const ListProvider = ({children, initialState}) => {
+    const [todos, dispatch] = useReducer(listReducer, initialState);
+
+     return(
+        <TodoListContext.Provider value={todos}>
+           <TodoListDispatchContext.Provider value={dispatch}>
+             {children}
+           </TodoListDispatchContext.Provider>
+        </TodoListContext.Provider>
+     )
+}
+function listReducer(state, action) {
```

Comment:
When you add an action type 'add' (add new todo), you'd need to pass the title of todo as well. Keeping it in mind, it's more common to have action parameter in the following format:
```
{
  type: actionType,
  payload: {} 
}
```
Where payload would contain any data that you may pass to the reduce function (e.g. action.payload.id in this case).

Reply:
Thank you !